rollouts/controllers/rollout_controller.go

```diff
@@ -378,11 +377,7 @@ func (r *RolloutReconciler) computeTargets(ctx context.Context,
 
 	// initially assume all the keys to be deleted
 	for _, rrs := range existingRRSs {
-		for _, cluster := range allowClusters {
-			if rrs.Spec.ClusterRef.Name == cluster.Name {
-				RRSkeysToBeDeleted[client.ObjectKeyFromObject(rrs)] = rrs
-			}
-		}
+		RRSkeysToBeDeleted[client.ObjectKeyFromObject(rrs)] = rrs
```

Comment:
interesting, now that we are computing targets at higher level, this is falling back in place.